internal/femto/femto.go

```diff
@@ -3,124 +3,145 @@ package femto
 
 import (
 	"encoding/json"
-	"errors"
 	"log/slog"
 	"net/http"
 	"sync/atomic"
+
+	"github.com/gpuctl/gpuctl/internal/types"
 )
 
 type Femto struct {
 	mux   http.ServeMux
 	reqNo atomic.Uint64
 }
 
-func (f *Femto) ServeHTTP(w http.ResponseWriter, r *http.Request) {
-	f.mux.ServeHTTP(w, r)
+type HTTPResponseContent[T any] struct {
+	Headers map[string]string
+	Body    T
+	Status  int
 }
 
-func OnPost[T any, R any](f *Femto, pattern string, handle PostFuncPure[T, R]) {
-	f.mux.HandleFunc(pattern, func(w http.ResponseWriter, r *http.Request) {
-		doPost(f, w, r, handle)
+type PostFunc[T any] func(T, *http.Request, *slog.Logger) (*HTTPResponseContent[types.Unit], error)
+type GetFunc[T any] func(*http.Request, *slog.Logger) (*HTTPResponseContent[T], error)
+
+func (femto *Femto) ServeHTTP(w http.ResponseWriter, r *http.Request) {
+	femto.mux.ServeHTTP(w, r)
```

Comment:
Single character receivers are idiomatic in Go. https://go.dev/wiki/CodeReviewComments#receiver-names. Changing this doesn't improve clairity, but does make the git diff more confusing to review.

Reply:
I think they're idiomatic so long as there are no more than two of them, and they are the same within the codebase. `Request` and `Response` need different character names then.